### PR TITLE
Message comparison in FilteringReporter strips trailing [quickfixable]

### PR DESCRIPTION
**Pull request by lrytz**

### CODE REVIEW

**Comment by lrytz** [comment]
2 steps forward (by som), one back (by me)

**Comment by som-snytt** [comment]
maybe try putting your shoes on backward.

File: src/compiler/scala/tools/nsc/reporters/Reporter.scala [hunk 1]
```diff
@@ -150,12 +124,17 @@ abstract class FilteringReporter extends Reporter {
...
+        messages(fpos) ::= stripQuickfixable(msg)
       }
       show
     }
   }
 
+  private def stripQuickfixable(msg: String): String = {
+    val i = msg.indexOf(" [quickfixable]")
+    if (i > 0) msg.substring(0, i) else msg
+  }
```
Comment (som-snytt) [inline src/compiler/scala/tools/nsc/reporters/Reporter.scala:136]: I would have `if (msg.endsWith(" [qfa]")) etc`.

**Review by som-snytt — APPROVED** [review]
Well that worked out. I just noticed javac prefixes the message as you had done, probably to make it easier to machine process?
```
Y.java:21: warning: [removal] ContentSignerParameters in com.sun.jarsigner has been deprecated and marked for removal
```
Maybe my issue with it was the length of `[quickfixable]`; `[fixable]` is the same length as `[removal]`. In ancient times, I remember log files with left-side columns that aligned nicely for visual scanning and machine parsing, `ERR: com.acme: uhoh`, or I don't remember exactly except that alignment mattered.

**Comment by som-snytt** [comment]
I updated the title to look more like the problem.

File: src/compiler/scala/tools/nsc/reporters/Reporter.scala (hunk 1 above)
Comment (lrytz) [reply src/compiler/scala/tools/nsc/reporters/Reporter.scala:136]: I noticed it's not always at the end, the "Scala 3 migration messages are errors under -Xsource:3..." appendix is added in RunReporting, after `[quickfixable]`.

Comment (som-snytt) [reply src/compiler/scala/tools/nsc/reporters/Reporter.scala:136]: ok I'm sorry I updated your title.

**Pull request merged**